### PR TITLE
fix: Build dockerfile correctly

### PR DESCRIPTION
**Pull request by felixwang9817**
Signed-off-by: Felix Wang <wangfelix98@gmail.com>

**What this PR does / why we need it**: #3223 introduced this `requirements.txt`. However, `docker build` copies files from the root directory, so the correct command would be `RUN pip install -r sdk/python/feast/infra/feature_servers/multicloud/requirements.txt` instead of `RUN pip install -r requirements.txt`. Instead of running this command, I decided to just remove the `requirements.txt` file and install the dependencies directly.

This bug is the reason why the Docker images were not built correctly during the latest 0.25.0 release.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/infra/feature_servers/multicloud/Dockerfile [hunk 1]
```diff
@@ -8,9 +8,8 @@ RUN apt update && \
         build-essential
 
 RUN pip install pip --upgrade
-COPY . .
+RUN pip install "feast[aws,gcp,snowflake,redis,go,mysql]"
```
Comment (achals) [inline sdk/python/feast/infra/feature_servers/multicloud/Dockerfile:11]: Hmmm won't this install the library from pypi, as opposed to a locally built wheel? Can that cause inconsistencies?

Comment (felixwang9817) [reply sdk/python/feast/infra/feature_servers/multicloud/Dockerfile]: fixed by restoring `COPY . .`

File: sdk/python/feast/infra/feature_servers/multicloud/Dockerfile [hunk 2]
```diff
@@ -9,8 +9,8 @@ RUN apt update && \
 
 RUN pip install pip --upgrade
 COPY . .
+RUN pip install ".feast[aws,gcp,snowflake,redis,go,mysql]"
```
Comment (achals) [inline sdk/python/feast/infra/feature_servers/multicloud/Dockerfile:12]: Do you need the period as well as `feast`?

Comment (felixwang9817) [reply sdk/python/feast/infra/feature_servers/multicloud/Dockerfile]: good catch

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**